tico/serialize/circle_graph.py

```diff
@@ -250,14 +245,8 @@ def add_tensor_from_scratch(
         if source_node is not None:
             self.name_to_node[tensor.name] = source_node
         tensor.shape = shape
-
-        # Handle dynamic shape
-        if any(isinstance(s, torch.SymInt) for s in tensor.shape):
-            tensor.shapeSignature = tensor.shape.copy()
-            for idx, s in enumerate(tensor.shape):
-                if isinstance(s, torch.SymInt):
-                    tensor.shape[idx] = 1
-                    tensor.shapeSignature[idx] = -1
+        if shape_signature:
```

Comment:
Similar to other suggestions.
```suggestion
        if shape_signature is not None:
```